### PR TITLE
fix(bigquery): import Mapping from collections.abc not from collections

### PR DESCRIPTION
**Pull request by alextford11**
Getting following deprecation warning in Bigquery:
```
File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/job.py", line 3216, in __iter__
    return iter(self.result())
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/job.py", line 3105, in result
    dest_table = Table(dest_table_ref, schema=schema)
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/table.py", line 335, in __init__
    self.schema = schema
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/table.py", line 400, in schema
    value = _to_schema_fields(value)
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/schema.py", line 284, in _to_schema_fields
    if not isinstance(field, (SchemaField, collections.Mapping)):
  File "/usr/lib/python3.7/collections/__init__.py", line 52, in __getattr__
    DeprecationWarning, stacklevel=2)
DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
```

### CODE REVIEW

File: bigquery/google/cloud/bigquery/schema.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 """Schemas for BigQuery tables / queries."""
 
-import collections
+from collections.abc import Mapping
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/schema.py]: import modules not classes (I know we're not consistent about this)

```suggestion
from collections import abc
```

File: bigquery/google/cloud/bigquery/schema.py [hunk 2]
```diff
@@ -281,7 +281,7 @@ def _to_schema_fields(schema):
         instance or a compatible mapping representation of the field.
     """
     for field in schema:
-        if not isinstance(field, (SchemaField, collections.Mapping)):
+        if not isinstance(field, (SchemaField, Mapping)):
```
Comment (tswast) [inline bigquery/google/cloud/bigquery/schema.py]:
```suggestion
        if not isinstance(field, (SchemaField, abc.Mapping)):
```

File: bigquery/google/cloud/bigquery/schema.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 """Schemas for BigQuery tables / queries."""
 
-import collections
+from collections import abc
```
Comment (samuelcolvin) [inline bigquery/google/cloud/bigquery/schema.py]:
```suggestion
try:
    from collections.abc import Mapping
except ImportError:
    # python 2.7
    from collections import Mapping
```

File: bigquery/google/cloud/bigquery/schema.py [hunk 4]
```diff
@@ -281,7 +281,7 @@ def _to_schema_fields(schema):
         instance or a compatible mapping representation of the field.
     """
     for field in schema:
-        if not isinstance(field, (SchemaField, collections.Mapping)):
+        if not isinstance(field, (SchemaField, abc.Mapping)):
```
Comment (samuelcolvin) [inline bigquery/google/cloud/bigquery/schema.py]:
```suggestion
        if not isinstance(field, (SchemaField, Mapping)):
```

File: bigquery/google/cloud/bigquery/schema.py (hunk 3 above)
Comment (tswast) [reply bigquery/google/cloud/bigquery/schema.py]: It's not that I "like" it, it's that this way it's easy to tell what module something came from.

We should use the `six` module, if the `abc` module is not available on 2.7.

**Comment by robert-l1** [comment]
I believe this update is causing issues with several versions of the BigQuery client class. My images are now throwing:
```
Traceback (most recent call last):
  File "test.py", line 20, in <module>
    from google.cloud import bigquery
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/__init__.py", line 35, in <module>
    from google.cloud.bigquery.client import Client
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/client.py", line 58, in <module>
    from google.cloud.bigquery import _pandas_helpers
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/_pandas_helpers.py", line 40, in <module>
    from google.cloud.bigquery import schema
  File "/usr/local/lib/python3.7/dist-packages/google/cloud/bigquery/schema.py", line 17, in <module>
    from six.moves import collections_abc
ImportError: cannot import name 'collections_abc' from 'six.moves' (unknown location)
```

I see the same in 2.7:
```File "test.py", line 20, in <module>
    from google.cloud import bigquery
  File "/usr/local/lib/python2.7/dist-packages/google/cloud/bigquery/__init__.py", line 35, in <module>
    from google.cloud.bigquery.client import Client
  File "/usr/local/lib/python2.7/dist-packages/google/cloud/bigquery/client.py", line 58, in <module>
    from google.cloud.bigquery import _pandas_helpers
  File "/usr/local/lib/python2.7/dist-packages/google/cloud/bigquery/_pandas_helpers.py", line 40, in <module>
    from google.cloud.bigquery import schema
  File "/usr/local/lib/python2.7/dist-packages/google/cloud/bigquery/schema.py", line 17, in <module>
    from six.moves import collections_abc
ImportError: cannot import name collections_abc
```

**Comment by tkrassowski** [comment]
@robert-l1
Upgrade your six package to newer version. Collections_abc support came in 1.13.1 (https://github.com/benjaminp/six/pull/241).

**Comment by neteler** [comment]
> @robert-l1
> Upgrade your six package to newer version. Collections_abc support came in 1.13.1 

FWIW: So far six 1.13.0 is the latest release (https://pypi.org/project/six/) and seems to be sufficient.

**Pull request merged**